packages/server/src/auth/CrossmintAuth.ts

```diff
@@ -21,6 +31,63 @@ export class CrossmintAuth {
         return new CrossmintAuth(crossmint);
     }
 
+    public getJWKSUri() {
+        return `${this.apiClient.baseUrl}/.well-known/jwks.json`;
+    }
+
+    public verifyCrossmintJwtToken(token: string) {
+        return verifyCrossmintJwtToken(token, this.getJWKSUri());
+    }
+
+    private async refreshAuthMaterial(refreshToken: string): Promise<AuthMaterial> {
+        const result = await this.apiClient.post(`api/${CROSSMINT_API_VERSION}/session/sdk/auth/refresh`, {
+            body: JSON.stringify({ refresh: refreshToken }),
+        });
+
+        const resultJson = (await result.json()) as AuthMaterialResponse;
+
+        return {
+            jwt: resultJson.jwt,
+            refreshToken: resultJson.refresh,
+            user: resultJson.user,
+        };
+    }
+
+    public async getSession(options: GenericRequest | AuthMaterialBasic): Promise<AuthSession> {
+        const { jwt, refreshToken } = "refreshToken" in options ? options : getAuthCookies(options);
+
+        if (!refreshToken) {
+            throw new CrossmintAuthenticationError("Refresh token not found");
+        }
+
+        try {
+            return await this.validateOrRefreshSession(jwt, refreshToken);
+        } catch (error) {
+            console.error("Failed to get session", error);
+            throw new CrossmintAuthenticationError("Failed to get session");
+        }
+    }
+
+    private async validateOrRefreshSession(jwt: string | undefined, refreshToken: string): Promise<AuthSession> {
+        if (jwt) {
+            try {
+                const decodedJwt = await this.verifyCrossmintJwtToken(jwt);
+                return {
+                    jwt,
+                    userId: decodedJwt.sub as string,
+                };
+            } catch (_) {
+                // The JWT is invalid, we need to refresh the session
+            }
+        }
+
+        const refreshedAuthMaterial = await this.refreshAuthMaterial(refreshToken);
+        return {
+            jwt: refreshedAuthMaterial.jwt,
+            userId: refreshedAuthMaterial.user.id,
+        };
+    }
```

Comment:
nit: keep public methods all first in the class, then private helpers. Should go:
1. `getSession` (public)
2. `getUser` (public)
3. `verifyCrossmintJWT` (public)
4. `getJWKSUri` (not sure yet)
5. `refreshAuthMaterial` (private)
6. `validateOrRefreshSession`